src/backend/cdb/dispatcher/cdbdisp_async.c

```diff
@@ -820,6 +825,39 @@ handlePollError(CdbDispatchCmdAsync *pParms)
 	return;
 }
 
+static void
+resetConnAndResult(CdbDispatchResult *dispatchResult)
+{
+	PGresult   *res;
+	PGnotify   *notify;
+	PGconn	   *conn = dispatchResult->segdbDesc->conn;
+
+	/* Free some memory in case we're tight on it. */
+	pqClearAsyncResult(conn);
+
+	/* Replace current result with a fatal error dummy one. */
+	conn->result = PQmakeEmptyPGresult(conn, PGRES_FATAL_ERROR);
+
+	/*
+	 * Discard anything that is unread. Since our result contains a fatal error,
+	 * we'll just consume the entire message without actually parsing it.
+	 */
+	conn->asyncStatus = PGASYNC_BUSY;
+
+	while ((res = PQgetResult(conn)) != NULL)
+		PQclear(res);
+
+	/* Free notices. */
+	while ((notify = PQnotifies(conn)) != NULL)
+		PQfreeNotify(notify);
```

Comment:
```suggestion
		PQfreemem(notify);
```
https://github.com/arenadata/gpdb/blob/84beaf4fd19ae4c30588c3c87342d10b16a1a5c0/src/interfaces/libpq/fe-exec.c#L3244-L3248